### PR TITLE
Fix GT ref and alt depth parsing when values are 0

### LINKED ISSUE #4272: Fix Junction Reads in fusion vcf
When no Junction Reads is present it currently shows -1 -> this should be "empty" instead

Update: other empty values are currently reported by 0. For consistency it would be good to keep all of them as 0, and maybe this will need to be update in the future (after the vcf has changed)

### PR DESCRIPTION
**Pull request by northwestwitch**
When you have the following genotype situation for a SV or fusion variant in the VCF:

`GT:DV:RV:FFPM	./1:0:0:0.0`

Then the variants gets parsed like this:

`{'individual_id': '888-88', 'display_name': '888-88', 'genotype_call': './1', 'so': None, 'alt_depth': -1, 'ref_depth': -1, 'read_depth': -1, 'alt_frequency': -1.0, 'genotype_quality': -1, 'ffpm': 0, 'split_read': 0}`

But instead we'd expect the alt_depth and read_depth to be 0

This is a fix (fix #4272)

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. Run scout setup demo on main branch and go to fusion variantS page, the last variant should have -1 as junction Reads

image

1. repeat with this branch and notice that the junction reads are 0

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [x] tests executed by CR

### CODE REVIEW

File: scout/parse/variant/genotype.py [hunk 1]
```diff
@@ -324,14 +326,14 @@ def get_ref_depth(
     """Get reference read depth"""
     ref_depth = int(variant.gt_ref_depths[pos])
     if ref_depth == -1:
-        if any([paired_end_ref, split_read_ref]):
```
Comment (northwestwitch) [inline scout/parse/variant/genotype.py:327]: The problem with this code is that any will be False even if one of these variables is present but is 0

File: scout/parse/variant/genotype.py [hunk 2]
```diff
@@ -324,14 +326,14 @@ def get_ref_depth(
...
+        if all(v is None for v in [paired_end_ref, split_read_ref]) is False:
```
Comment (northwestwitch) [inline scout/parse/variant/genotype.py:329]: This checks that there is at least one of the variables is not None instead

**Comment by northwestwitch** [comment]
test using this branch:  🆗 

image

File: scout/parse/variant/genotype.py [hunk 3]
```diff
@@ -324,24 +326,23 @@ def get_ref_depth(
...
+
+    REF_ITEMS_LIST: List[tuple] = [
+        (paired_end_ref, split_read_ref),
+        (spanning_ref, flanking_ref, inrepeat_ref),
+    ]
+
+    for list in REF_ITEMS_LIST:
+        if all(item is None for item in list) is True:
+            continue
+        alt_depth = 0
```
Comment (dnil) [inline scout/parse/variant/genotype.py:339]:
```suggestion
        ref_depth = 0
```

File: scout/parse/variant/genotype.py [hunk 4]
```diff
@@ -324,24 +326,23 @@ def get_ref_depth(
...
+        for item in list:
+            if item:
+                alt_depth += item
```
Comment (dnil) [inline scout/parse/variant/genotype.py:342]:
```suggestion
                ref_depth += item
```

File: scout/parse/variant/genotype.py [hunk 5]
```diff
@@ -358,32 +359,26 @@ def get_alt_depth(
...
+        alt_depth = int(variant.format("VD")[pos][0])
+
+    ALT_ITEMS_LIST: List[tuple] = [
+        (paired_end_alt, split_read_alt),
+        (clip5_alt, clip3_alt),
+        (spanning_alt, flanking_alt, inrepeat_alt),
+    ]
+
+    for list in ALT_ITEMS_LIST:
+        if all(item is None for item in list) is True:
```
Comment (dnil) [inline scout/parse/variant/genotype.py:375]:
```suggestion
        if all(item is None for item in list):
```

File: scout/parse/variant/genotype.py [hunk 6]
```diff
@@ -324,24 +326,23 @@ def get_ref_depth(
...
-            ref_depth += spanning_mei_ref
+        return
+
+    REF_ITEMS_LIST: List[tuple] = [
+        (paired_end_ref, split_read_ref),
+        (spanning_ref, flanking_ref, inrepeat_ref),
+    ]
+
+    for list in REF_ITEMS_LIST:
+        if all(item is None for item in list) is True:
```
Comment (dnil) [inline scout/parse/variant/genotype.py:337]:
```suggestion
        if all(item is None for item in list):
```

**Review by dnil — CHANGES_REQUESTED** [review]
Looks ok in principle, but there was a small copy-paste issue that I think prevented this from actually working. Which in turn probably means we should probably have a test for it? That might have helped the original issue as well. 😊

File: scout/parse/variant/genotype.py (hunk 3 above)
Comment (northwestwitch) [reply scout/parse/variant/genotype.py:339]: Ah, right, thanks. I'll write a test, it's a good idea!

File: tests/parse/test_parse_genotype.py [hunk 7]
```diff
@@ -55,3 +61,49 @@ def test_parse_cnvnator():
...
+def test_get_alt_depth_zeroes(one_variant: "Cyvcf2Variant"):
+    """Make sure get_alt_depth returns 0 when provided parameters are all 0."""
+
+    # GIVEN a sample at a given position 2 in the VCF
+    position = 0
+    # GIVEN a variant which has alt_depth == 0 for that individual
+    assert one_variant.gt_alt_depths[position] == 0
+
+    # WHEN alt depth is computed using get_alt_depth function with null parameters
+    alt_depth: int = get_ref_depth(
```
Comment (dnil) [inline tests/parse/test_parse_genotype.py:97]:
```suggestion
    alt_depth: int = get_alt_depth(
```

Comment (dnil) [reply tests/parse/test_parse_genotype.py:97]: And so on.. 😊

**Review by dnil — COMMENTED** [review]
Thank you for adding the tests! Just a couple of details. 😂

**Comment by northwestwitch** [comment]
> Thank you for adding the tests! Just a couple of details. 😂

Right, they should disable copy and paste on my keyboard!

File: scout/parse/variant/genotype.py [hunk 8]
```diff
@@ -358,32 +359,26 @@ def get_alt_depth(
...
-
-        if any([spanning_alt, flanking_alt, inrepeat_alt]):
-            alt_depth = 0
-            if spanning_alt:
-                alt_depth += spanning_alt
-            if flanking_alt:
-                alt_depth += flanking_alt
-            if inrepeat_alt:
-                alt_depth += inrepeat_alt
+    if alt_depth != -1:
```
Comment (northwestwitch) [inline scout/parse/variant/genotype.py:362]: And his is != -1:

File: scout/parse/variant/genotype.py [hunk 9]
```diff
@@ -324,24 +326,23 @@ def get_ref_depth(
     """Get reference read depth"""
     ref_depth = int(variant.gt_ref_depths[pos])
     if ref_depth == -1:
```
Comment (northwestwitch) [inline scout/parse/variant/genotype.py:328]: This is weird, why is this == -1

Comment (dnil) [reply scout/parse/variant/genotype.py:328]: Since "." in the GT call?

File: scout/parse/variant/genotype.py (hunk 8 above)
Comment (dnil) [reply scout/parse/variant/genotype.py:362]:
```suggestion
    if alt_depth == -1:
```
Sry, missed that one!

Comment (dnil) [reply scout/parse/variant/genotype.py:362]: Return on that one, so this would be the correct one.. 🙄

File: scout/parse/variant/genotype.py (hunk 9 above)
Comment (dnil) [reply scout/parse/variant/genotype.py:328]:
```suggestion
    if ref_depth != -1:
```

Comment (dnil) [reply scout/parse/variant/genotype.py:328]: Sry, didn't think about the return at first.

File: scout/parse/variant/genotype.py (hunk 8 above)
Comment (northwestwitch) [reply scout/parse/variant/genotype.py:362]: Sorry I son't understand. Shouldn't it return alt_depth if alt_depth != -1?

Comment (dnil) [reply scout/parse/variant/genotype.py:362]: Ahem yes? Oh.. Actually you do need copy paste! 😜

File: scout/parse/variant/genotype.py [hunk 10]
```diff
@@ -358,32 +359,26 @@ def get_alt_depth(
...
-        if any([spanning_alt, flanking_alt, inrepeat_alt]):
-            alt_depth = 0
-            if spanning_alt:
-                alt_depth += spanning_alt
-            if flanking_alt:
-                alt_depth += flanking_alt
-            if inrepeat_alt:
-                alt_depth += inrepeat_alt
+    if alt_depth != -1:
+        return
```
Comment (dnil) [inline scout/parse/variant/genotype.py:363]:
```suggestion
        return alt_depth
```

**Comment by northwestwitch** [comment]
Bah, I'm a bit tired just now and I feel I just going to mess up more. I'll continue tomorrow with this one!

File: scout/parse/variant/genotype.py [hunk 11]
```diff
@@ -323,25 +325,24 @@ def get_ref_depth(
...
-                ref_depth += spanning_ref
-            if flanking_ref:
-                ref_depth += flanking_ref
-            if inrepeat_ref:
-                ref_depth += inrepeat_ref
-
-        if spanning_mei_ref:
-            ref_depth += spanning_mei_ref
+    if ref_depth != -1:
+        return
```
Comment (dnil) [inline scout/parse/variant/genotype.py:329]:
```suggestion
        return ref_depth
```

**Comment by dnil** [comment]
> Bah, I'm a bit tired just now and I feel I just going to mess up more. I'll continue tomorrow with this one!

Sounds like a plan! I should also be homeward bound. See you tomorrow!

**Comment by northwestwitch** [comment]
> > Bah, I'm a bit tired just now and I feel I just going to mess up more. I'll continue tomorrow with this one!
> 
> Sounds like a plan! I should also be homeward bound. See you tomorrow!

yes, thanks for the review!

**Review by dnil — APPROVED** [review]
👍 

And if you are worried about zero depth from earlier in the flow, I would suggest rather to work at the place where the zero is set instead of the -1! 😊

**Pull request merged**